libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverview.tsx

```diff
@@ -657,27 +657,47 @@ export class ModelOverview extends React.Component<
           new AbortController().signal
         )
         .then((result) => {
-          // Assumption: the lengths of `result` and `selectionIndexes` are the same.
+          const [allCohortMetrics, cohortClasses] = result;
```

Comment:
it looks like there is a breaking change here in the UI - how can we ensure that the new UI with the older saved dashboards/python dependencies still works here and doesn't fail?

Reply:
Perhaps the env can be rebuilt which installs the latest version of `responsibleai_vision` and the DPv2 components can rerun. Have addressed this in more detail outside this PR. This optimization is almost a necessary improvement because of the way the torchmetrics API is invoked, and considering the computational cost of calculating object detection metrics. Let me know your suggestions, thanks a lot!

Reply:
Update: Confirmed it's okay to go with this change.